packages/template-retail-react-app/app/hooks/use-derived-product.js

```diff
@@ -38,7 +39,35 @@ export const useDerivedProduct = (
         isProductPartOfBundle
     )
     const [quantity, setQuantity] = useState(initialQuantity)
+    const {site} = useMultiSite()
 
+    // Helper function to get inventory by ID
+    const getInventoryById = (inventoryId) => {
+        if (!inventoryId || !product?.inventories) {
+            return null
+        }
+        return product.inventories.find((inv) => inv.id === inventoryId)
+    }
+
+    const selectedStore = (() => {
+        try {
+            if (typeof window !== 'undefined' && site?.id) {
+                return JSON.parse(window.localStorage.getItem(`store_${site.id}`))
+            }
+        } catch (e) {
+            // intentionally empty: ignore errors
+        }
+        return null
+    })()
+
+    const selectedStoreInventory = getInventoryById(selectedStore?.inventoryId)
+    const selectedStoreStockLevel = selectedStoreInventory?.stockLevel || 0
+    // selectedStoreStockLevel and selectedStoreInventory are already variant specific,
+    // so we don't need to check for variation attributes
+    const isSelectedStoreOutOfStock =
```

Comment:
Maybe something to be considered is the treatment of setproducts, will the use-derived product determine the inventory of all the set children correctly

There is a WI for that W-18751492, we should validate the scenario there